src/en/flamecomics/src/eu/kanade/tachiyomi/extension/en/flamecomics/FlameComics.kt

```diff
@@ -4,62 +4,172 @@ import android.graphics.Bitmap
+            }
+        }
+        page--
 
-        if (!hasSplitImages) {
-            return super.pageListParse(document)
+        var lastPage = page * 20 + 20
+        if (lastPage > manga.size) {
+            lastPage = manga.size
         }
+        if (lastPage < 0) lastPage = 0
+        return MangasPage(manga.subList(page * 20, lastPage), lastPage < manga.size)
+    }
 
-        return document.select("#readerarea p:has(img), $composedSelector").toList()
-            .filter {
-                it.select("img").all { imgEl ->
-                    imgEl.attr("abs:src").isNullOrEmpty().not()
-                }
-            }
-            .mapIndexed { i, el ->
-                if (el.tagName() == "p") {
-                    Page(i, "", el.select("img").attr("abs:src"))
-                } else {
-                    val imageUrls = el.select("img")
-                        .joinToString("|") { it.attr("abs:src") }
-
-                    Page(i, document.location(), imageUrls + COMPOSED_SUFFIX)
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        GET(
+            baseUrl.toHttpUrl().newBuilder().apply {
+                addPathSegment("browse")
+                addQueryParameter("search", query)
+                addQueryParameter("page", page.toString())
+            }.build(),
+            headers,
+        )
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET(
+            baseUrl.toHttpUrl().newBuilder().apply {
+                addPathSegment("browse")
+                addQueryParameter("page", page.toString())
+            }.build(),
+            headers,
+        )
+
+    override fun latestUpdatesRequest(page: Int): Request = popularMangaRequest(page)
+
+    override fun mangaDetailsParse(response: Response): SManga = SManga.create().apply {
+        val seriesData = getJsonData<MangaPageData>(response.asJsoup())?.props?.pageProps?.series
+            ?: return SManga.create()
+
+        title = seriesData.title
+        thumbnail_url = cdn.toHttpUrl().newBuilder().apply {
+            addPathSegment("series")
+            addPathSegment(seriesData.series_id.toString())
+            addPathSegment(seriesData.cover)
+        }.build().toString()
+        description = seriesData.description
+        author = seriesData.author
+        status = when (seriesData.status.lowercase()) {
+            "ongoing" -> SManga.ONGOING
+            "dropped" -> SManga.CANCELLED
+            "hiatus" -> SManga.ON_HIATUS
+            "completed" -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val mangaPageData = getJsonData<MangaPageData>(response.asJsoup()) ?: return listOf()
+        return mangaPageData.props.pageProps.chapters.map { chapter ->
+            SChapter.create().apply {
+                setUrlWithoutDomain(
```

Comment:
Regex you can use to get seriesid/token from user-facing URLs:
```kt
private val patternMangaUrl = """/series/(?<seriesId>[0-9]+)(?:/(?<token>[0-9A-Za-z]+))?""".toRegex()
```

Reply:
The chapters appear to each have an `id`. I could try to use them since they seem more permanent than the tokens or would that have the same problem as with `chapterNum`?

Reply:
I don't see `chapterNum` as unique enough. You have special chapters, multiple groups, etc. For `tokens`- I think it should be safe to rely on since the public URL is shown with this token, and people might bookmark it. Meaning they should maintain compatibility with it in the future.